src/unix.rs

```diff
@@ -102,10 +104,16 @@ fn localized_msg_fmt(msg: &[u8], f: &mut Formatter) -> fmt::Result {
 }
 
 pub fn errno_fmt(e: i32, f: &mut Formatter) -> fmt::Result {
-    let msg = unsafe {
-        let msg = strerror(e) as *const c_char;
-        slice::from_raw_parts(msg as *const u8, strlen(msg))
-    };
+    // 128 bytes should be long enough for all error messages
+    const BUF_SIZE: usize = 128;
+
+    let mut buf: [c_char; BUF_SIZE] = [0; BUF_SIZE];
+    if unsafe { strerror_r(e, buf.as_mut_ptr(), BUF_SIZE) } < 0 {
```

Comment:
This check is not absolutely correct since glibc 2.13 according to the `strerror_r` manpage. It would be better to use `!= 0` instead of `< 0`.

Reply:
You’re absolutely right, I just copied this from `std` which means they have this bug too 🤔 

I also had to modify one of the tests so that only valid errno values are tested, since before it seems that invalid errno values were being silently swallowed.